apps/examples/testcase/le_tc/filesystem/itc_fs.c

```diff
@@ -0,0 +1,1442 @@
+#include <tinyara/config.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <dirent.h>
+#include <fcntl.h>
+#include <string.h>
+#include <sys/mount.h>
+#include <sys/stat.h>
+#include "tc_common.h"
+
+/****************************************************************************
+ * Definitions
+ ****************************************************************************/
+
+#ifdef CONFIG_FS_SMARTFS
+
+#if defined(CONFIG_SIDK_S5JT200_AUTOMOUNT_USERFS)
+#define MOUNT_DEV_DIR CONFIG_SIDK_S5JT200_AUTOMOUNT_USERFS_DEVNAME
+
+#elif defined(CONFIG_ARTIK05X_AUTOMOUNT_USERFS)
+#define MOUNT_DEV_DIR CONFIG_ARTIK05X_AUTOMOUNT_USERFS_DEVNAME
+
+#else
+#define MOUNT_DEV_DIR "/dev/smart1"
+#endif
+
+#define TARGET_FS_NAME "smartfs"
+#else
+#define MOUNT_DEV_DIR "/dev/fs1"
+#define TARGET_FS_NAME "unknwon"
+#endif
+
+#define BUFFER_LEN 30
+#define MOUNT_DIR CONFIG_MOUNT_POINT
+#define VFS_FILE_PATH MOUNT_DIR"vfs"
+#define VFS_FOLDER_PATH MOUNT_DIR"folder"
+#define VFS_LOOP_COUNT 5
+#define VFS_TEST_CONTENTS_1 "THIS IS VFS TEST 1"
+#define VFS_TEST_CONTENTS_2 "THIS IS VFS TEST 2"
+
+/**
+* @fn               add_dirs
+* @description      add directory VFS_FOLDER_PATH, and count of inner directories
+* @return           void
+*/
+static void add_dirs(int count)
+{
+	char folder_name[BUFFER_LEN];
+	int len;
+	int i;
+	strcpy(folder_name, VFS_FOLDER_PATH);
+	strcat(folder_name, "/0\0");
+	len = strlen(folder_name);
+	mkdir(VFS_FOLDER_PATH, 0777);
+	for (i = 0; i < count; i++) {
+		folder_name[len - 1] = '0' + i;
+		mkdir(folder_name, 0777);
+	}
+}
+
+/**
+* @fn               remove_dirs
+* @description      remove directory VFS_FOLDER_PATH, and count of inner directories
+* @return           void
+*/
+static void remove_dirs(int count)
+{
+	char folder_name[BUFFER_LEN];
+	int len;
+	int i;
+	strcpy(folder_name, VFS_FOLDER_PATH);
+	strcat(folder_name, "/0\0");
+	len = strlen(folder_name);
+	for (i = 0; i < count; i++) {
+		folder_name[len - 1] = '0' + i;
+		rmdir(folder_name);
+	}
+	rmdir(VFS_FOLDER_PATH);
+}
+
+/**
+* @testcase         itc_fs_vfs_umount_n_twice
+* @brief            Unmount file system
+* @scenario         Unmount file system twice.
+* @apicovered       umount
+* @precondition     File system should be mounted.
+* @postcondition    NA
+*/
+static void itc_fs_vfs_umount_n_twice(void)
+{
+	int ret;
+
+	ret = umount(CONFIG_MOUNT_POINT);
+	TC_ASSERT_EQ("umount", ret, OK);
+
+	ret = umount(CONFIG_MOUNT_POINT);
+	TC_ASSERT_EQ("umount", ret, ERROR);
+
+	TC_SUCCESS_RESULT();
+}
+
+/**
+* @testcase         itc_fs_vfs_mount_p_read_mode
+* @brief            Mount file system
+* @scenario         Mount file system in read mode, and try to create directory.
+* @apicovered       mount
+* @precondition     File system should be initialized and not mounted.
+* @postcondition    NA
+*/
+static void itc_fs_vfs_mount_p_read_mode(void)
+{
+	int ret;
+
+	ret = mount(MOUNT_DEV_DIR, CONFIG_MOUNT_POINT, TARGET_FS_NAME, MS_RDONLY, NULL);

```

Comment:
mount function use FAR const char *target as a second parameter, you'd better rename TARGET_FS_NAME to FS_TYPE